File: bioskopina_mobile/lib/models/user_role.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/role.dart';
part 'user_role.g.dart';

@JsonSerializable()
class UserRole {
  int? id;
  int? userId;
  int? roleId;
  bool? canReview;
  bool? canAskQuestions;
  bool? canParticipateInClubs;
  Role? role;

  UserRole(
    this.id,
    this.userId,
    this.roleId,
    this.canReview,
    this.canAskQuestions,
    this.canParticipateInClubs,
    this.role,
  );

  factory UserRole.fromJson(Map<String, dynamic> json) =>
      _$UserRoleFromJson(json);

  Map<String, dynamic> toJson() => _$UserRoleToJson(this);
}

File: backend/Bioskopina.Model/UserRole.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bioskopina.Model
{
    public class UserRole
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public int RoleId { get; set; }

        public bool CanReview { get; set; }

        public bool CanAskQuestions { get; set; }

        public bool? CanParticipateInClubs { get; set; }

        public virtual Role Role { get; set; } = null!;

    }
}
